File: question1.pl
/* Dawson Black */
/* 1.3 */

foo(X,Y,Z) :- Y =< 0, Z is X, !.
foo(X,Y,Z) :- X =< 0, Z is Y, !.
foo(X,Y,Z) :- X < Y,
X1 is X-2,
Y1 is Y-3,
(foo(X1,Y1,Z1)),
Z is Y+Z1;
X1 is X-1,
Y1 is Y-2,
(foo(X1,Y1,Z1)),
Z is X+Z1.

File: question1.cpp
// Dawson Black
// 1.1

#include <iostream>
using namespace std;

int foo(int x, int y) // size-n
{
	if (y <= 0) // stopping condition
	{
		return x;
	}
	else if (x <= 0) // stoppiong condition
	{
		return y;
	}
	else if (x >= y) // size-m
	{
		return x + foo(x - 1, y - 2); // constructs size-n out of size-m
	}
	else if (x < y) // size-m
	{
		return y + foo(x - 2, y - 3); // constructs size-n out of size-m
	}
}

int main()
{
	int x, y;

	cout << "Enter an integer: " << endl;
	cin >> x;
	cout << "\nEnter a second integer: " << endl;
	cin >> y;
	cout << "\nThe value of foo(" << x << "," << y << ") is " << foo(x, y) << "." << endl;
}